include/bitcoin/bitcoin/settings.hpp

```diff
@@ -36,6 +36,15 @@ class BC_API settings
     settings();
     settings(config::settings context);
 
+    // Currency unit helper methods (uint64_t).
+    //--------------------------------------------------------------------------
+
+    uint64_t max_money_recursive(uint64_t money) const;
+    uint64_t bitcoin_to_satoshi(uint64_t bitcoin_uints=1) const;
+    uint64_t initial_block_subsidy_satoshi() const;
+    uint64_t subsidy_interval(bool retarget=true) const;
+    uint64_t max_money(bool retarget=true) const;
```

Comment:
Logic should be kept out of the settings classes. I've only used this approach for simple data type conversions (e.g. integer to time). These functions can remain centralized or moved to class static helpers.

Reply:
I feel the same about logic inside the settings class. That's why I explicitly asked you on slack about where these functions should go before starting to work on this PR. You've told me that it's ok for now to move them into the settings class :man_shrugging:

Reply:
I suppose now is no longer then :?